### PR TITLE
Enable all static-checks in golangci-lint

### PR DESCRIPTION
**Pull request by Madhu-1**
# Describe what this PR does #

* Enable all static-checks in golangci-lint
* Update golangci-lint version
* Fix issue found in golangci-lint

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@humblec PTAL this will catch more static issues in the code

File: cmd/cephcsi.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func getType() string {
 
 func getDriverName() string {
 	// was explicitly passed a driver name
-	if driverName != nil && len(*driverName) != 0 {
+	if driverName != nil && *driverName != "" {
```
Comment (humblec) [inline cmd/cephcsi.go]: @Madhu-1 length check (len(..) )looks to be the better method

File: e2e/utils.go [hunk 2]
```diff
@@ -137,8 +137,8 @@ func getMons(ns string, c kubernetes.Interface) []string {
 	svcList, err := c.CoreV1().Services(ns).List(opt)
 	Expect(err).Should(BeNil())
 	services := make([]string, 0)
-	for _, svc := range svcList.Items {
-		s := fmt.Sprintf("%s.%s.svc.cluster.local:%d", svc.Name, svc.Namespace, svc.Spec.Ports[0].Port)
+	for i := range svcList.Items {
```
Comment (humblec) [inline e2e/utils.go]: @Madhu-1 svc is more readable and meaningful, Isnt it ?

File: pkg/csi-common/driver.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ type CSIDriver struct {
 // NewCSIDriver Creates a NewCSIDriver object. Assumes vendor
 // version is equal to driver version &  does not support optional
 // driver plugin info manifest field. Refer to CSI spec for more details.
-func NewCSIDriver(name string, v string, nodeID string) *CSIDriver {
+func NewCSIDriver(name, v, nodeID string) *CSIDriver {
```
Comment (humblec) [inline pkg/csi-common/driver.go]: @Madhu-1 in `nodeUnPublishVolume()` in mountcache above, we removed this explicit mention of variables, but we are bringing it here :), we need to be consistent :)

File: pkg/csi-common/utils.go [hunk 4]
```diff
@@ -64,11 +64,11 @@ func NewDefaultControllerServer(d *CSIDriver) *DefaultControllerServer {
 }
 
 // NewControllerServiceCapability returns controller capabilities
-func NewControllerServiceCapability(cap csi.ControllerServiceCapability_RPC_Type) *csi.ControllerServiceCapability {
+func NewControllerServiceCapability(conCap csi.ControllerServiceCapability_RPC_Type) *csi.ControllerServiceCapability {
 	return &csi.ControllerServiceCapability{
 		Type: &csi.ControllerServiceCapability_Rpc{
 			Rpc: &csi.ControllerServiceCapability_RPC{
-				Type: cap,
+				Type: conCap,
```
Comment (humblec) [inline pkg/csi-common/utils.go]: I would suggest `ctrlCap` instead of `conCap`

File: pkg/util/nodecache.go [hunk 5]
```diff
@@ -59,9 +59,9 @@ func (nc *NodeCache) ForAll(pattern string, destObj interface{}, f ForAllFunc) e
 	if err != nil {
 		return errors.Wrapf(err, "node-cache: failed to read %s folder", nc.BasePath)
 	}
-	path := path.Join(nc.BasePath, nc.CacheDir)
+	cPath := path.Join(nc.BasePath, nc.CacheDir)
```
Comment (humblec) [inline pkg/util/nodecache.go]: May be `cachePath` ?

File: cmd/cephcsi.go (hunk 1 above)
Comment (Madhu-1) [reply cmd/cephcsi.go]: issue found by golangci-lint

File: e2e/utils.go (hunk 2 above)
Comment (Madhu-1) [reply e2e/utils.go]: same here, this was the issue found in static check

File: pkg/csi-common/driver.go (hunk 3 above)
Comment (Madhu-1) [reply pkg/csi-common/driver.go]: this PR just covert `(name string, v string, nodeID string)`  to  `(name , v , nodeID string)

File: pkg/csi-common/utils.go (hunk 4 above)
Comment (Madhu-1) [reply pkg/csi-common/utils.go]: done

File: pkg/util/nodecache.go (hunk 5 above)
Comment (Madhu-1) [reply pkg/util/nodecache.go]: done

**Comment by Madhu-1** [comment]
@humblec PTAL

**Comment by humblec** [comment]
LGTM. Thanks .

**Pull request merged**